code/modules/food_and_drinks/food/condiment.dm

```diff
@@ -103,20 +129,13 @@
 			icon_state = temp_list[1]
 			name = temp_list[2]
 			desc = temp_list[3]
-
```

Comment:
You should move all this into `update_icon` since it's integral part of it, it shouldn't be split like that.

Reply:
ok i did it